File: src/main/java/com/project/isa/model/Offer.java
package com.project.isa.model;

import javax.persistence.*;

import com.project.isa.model.auth.Supplier;

import java.util.Date;

//Svaki dobavljac moze da daje ponudu za svaku porudzbenicu ciji sadrzaj ima na stanju-navodi cenu i rok isporuke
@Entity
@Table(name="offers")
public class Offer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Order order;

    @ManyToOne
    private Supplier supplier;

    @Column(name="price")
    private Integer price;

    @Column(name="delivery_date")
    private Date deliveryDate;
}

File: api/src/entities/Offer.ts
import { Field } from "type-graphql";
import { BaseEntity, Column, CreateDateColumn, Entity, ManyToOne, PrimaryGeneratedColumn, UpdateDateColumn } from "typeorm";
import { Order } from "./Order";
import { User } from "./User";

@Entity()
export class Offer extends BaseEntity{

    @PrimaryGeneratedColumn()
    id!: number;

    @ManyToOne(()=>User)
    supplier: User;

    @ManyToOne(()=>Order)
    order: Order;

    @Field()
    @Column()
    state: string;

    @Column()
    price: number;

    @Column({type :Date})
    deliveryDate: Date;

    @CreateDateColumn()
    createdAt = new Date();

    @UpdateDateColumn()
    updatedAt = new Date();

}